server/src/utils/MixerConnection.ts

```diff
@@ -190,13 +197,19 @@ export class MixerGenericConnection {
 
         state.channels[0].chMixerConnection.forEach(
             (chMixerConnection: IchMixerConnection, mixerIndex: number) => {
-                chMixerConnection.channel.forEach(
-                    (channel: IChannel, channelIndex: number) => {
-                        if (faderIndex === channel.assignedFader) {
-                            this.fadeInOut(mixerIndex, channelIndex, fadeTime)
+                if (mixerIndex !== allMixersButThis) {

```

Comment:
You have a lot of layers of nesting here. That, plus you are in a foreach within a foreach makes this piece of code very complex.

You can reduce the complexity a lot just by reverting the ifs and extract to smaller methods, but keeping the logic:
```
updateOutLevel = (...) => {
  fadeTime = getFadeTimeFromState(fadeTime)
  state.channels[0].chMixerConnection.foreach((...) => {
    if (mixerIndex === allMixersButThis) {
      return
    }
    fadeChannelsOnMixerConnection(...)
  }
}

getFadeTimeFromState = (...) => {
  // The fadeTime logic that currently is in updateOutLevel goes here
}

fadeChannelsOnMixerConnection = (mixerConnection, ...) => {
  mixerConnection.channel.forEach((...) => {
    if (fadeIndex !== channel.assignedFader) {
      return
    }
    this.fadeInOut(...)
  }
}
```

Reply:
Agree. As noted it's made with as little changes as possible.
A refactor should be made, so the reference is iterating over state.fader[faderIndex].assignedChannels
and assignedChannels should be rewritten so it was: {mixerIndex: xx, channel: xx}

This would give a huge performance boost, but it's a large refactor.